src/main/java/duke/tasklist/TaskList.java

```diff
@@ -0,0 +1,220 @@
+package duke.tasklist;
+import duke.Ui;
+import duke.dukeexceptions.MissingArgumentException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+/**
+ * Stores tasks in a list
+ */
+public class TaskList {
+    protected Task[] list;
+    protected int len;
+
+    /**
+     * Constructor for the task list.
+     */
+    public TaskList() {
+        this.list = new Task[100];
+        this.len = 0;
+    }
+
+    /**
+     * Retrieves a task from the task list.
+     *
+     * @param index index of the task you want to retrieve
+     * @return the task
+     */
+    public Task showTask(int index) {
+        return list[index];
+    }
+
+    /**
+     * Prints out all the task in the task list.
+     */
+    public void list() {
+        String result = "";
+        for (int i = 0; i < this.len; i++) {
+            int index = i + 1;
+            result += "  " + index + ". " + list[i].toString();
+        }
+        System.out.print(result);
```

Comment:
Is there a way to allow the Ui class to handle all user input and output instead of printing from this class?